### PR TITLE
Update api

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -51,41 +34,48 @@ constants.CONTEXTS[constants.SECURITY_CONTEXT_V2_URL] = JSON.parse(
...
-  label: 'Edit Identity',
-  comment: 'Required to edit an Identity.'
-};
-permissions.IDENTITY_UPDATE_MEMBERSHIP = {
-  id: 'IDENTITY_UPDATE_MEMBERSHIP',
-  label: 'Update Membership',
-  comment: 'Required to update membership in groups.'
+permissions.IDENTITY_META_UPDATE = {
+  id: 'IDENTITY_META_UPDATE',
+  label: 'Update Identity Meta',
```
Comment (davidlehn) [inline lib/config.js:64]: spell out as "Metadata"?

**Comment by mattcollier** [comment]
github is not showing index.js so I can't comment there directly.  Couple notes from the linter:
If the function body starting on L73 is indented, then eslint is happy and no wrapping needs to be done.

on L646, it's notifying that `roleId` could be a `const` since it is never reassigned.

In 10-api test, the global `should` should be in the .eslintrc file (copy from bedrock-ledger-*) and not defined in comment at the top of file.  Too many indent issues to mention here.

**Comment by mattcollier** [comment]
test.js was not modified here, but `var` is still in use there.

**Comment by dlongley** [comment]
> In 10-api test, the global should should be in the .eslintrc file (copy from bedrock-ledger-*) ...

I couldn't find that file `.eslintrc` in bedrock-ledger-node and bedrock-ledger-consensus-continuity. Where should it be?

**Comment by mattcollier** [comment]
https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/blob/master/test/mocha/.eslintrc

File: lib/index.js [hunk 2]
```diff
@@ -1,497 +1,730 @@
...
+    id: database.hash(id),
+    'meta.sequence': sequence
+  }, {
+    $set: {identity: patched},
+    $inc: {'meta.sequence': 1}
+  }, database.writeOptions);
+
+  if(result.result.n === 0) {
+    return new BedrockError(
+      'Could not update Identity. Record sequence does not match.',
```
Comment (davidlehn) [inline lib/index.js:409]: Could result.n be zero in other cases than this?

Comment (dlongley) [reply lib/index.js:409]: Not at present.

File: lib/config.js (hunk 1 above)
Comment (dlongley) [reply lib/config.js:64]: Nah

**Comment by davidlehn** [comment]
Looks ok to me, though I'm not sure how the breaking changes effect other libs.  I'm guessing any issues with that will be addressed moving forward.

**Review by davidlehn — APPROVED** [review]

**Comment by dlongley** [comment]
@mattcollier,

Ready for another pass.

**Comment by dlongley** [comment]
@mattcollier,

> If the function body starting on L73 is indented, then eslint is happy and no wrapping needs to be done.

Not sure what this means ... style seems fine to me as is. It can't fit on one line so it has to be broken into two.

**Comment by mattcollier** [comment]
I'm saying that eslint would like L73-113 to be indented.  Even if you prefer not to configure your editor for use with eslint, you could run the command line version.

**Review by mattcollier — APPROVED** [review]
Awesome!

**Comment by dlongley** [comment]
@mattcollier,

If we're going to change `PermissionDenied` to `NotAllowedError` then I want to make those changes before I pull this in an release version 5. Are we keeping `PermissionedDenied` or changing it?

**Pull request merged**